File: 0090_Subsets II/SubsetsII.cs
public class Solution {
    public IList<IList<int>> SubsetsWithDup(int[] nums) {
        Array.Sort(nums);
        var ans = new List<IList<int>>();
        DFS(nums, 0, new List<int>(), ans, new HashSet<string>());
        return ans;
    }
    
    private void DFS(int[] nums, int index, IList<int> list, IList<IList<int>> ans, HashSet<string> set)
    {
        var result = String.Join(", ", list.ToArray());
        if(!set.Contains(result))  
        {
            ans.Add(new List<int>(list));
            set.Add(result);
        }
        
        for(int i=index;i<nums.Length;i++)
        {
            list.Add(nums[i]);
            DFS(nums, i+1, list, ans, set);
            list.RemoveAt(list.Count-1);
        }
    }
}

File: 0090_Subsets II/SubsetsII.py
class Solution(object):
    def subsetsWithDup(self, nums):
        def DFS(nums, n, s, curr, ans):
            if len(curr) == n:
                ans.append(curr[:])
                return
            for i in range(s,len(nums)):
                if i > s and nums[i] == nums[i-1]:continue
                curr.append(nums[i])
                DFS(nums, n, i+1, curr, ans)
                curr.pop()
        
        ans = []
        nums.sort()
        for n in range(0, len(nums)+1):
            DFS(nums, n,0, [],ans)
        
        return ans
        